avocado/utils/kernel.py

```diff
@@ -152,22 +153,33 @@ def configure(self, targets=('defconfig'), extra_configs=None):
                        config_file.name]
                 process.run(" ".join(cmd), shell=True)
 
-    def build(self, binary_package=False):
+    def build(self, binary_package=False, njobs=1):
         """
         Build kernel from source.
 
         :param binary_package: when True, the appropriate
                                   platform package is built
                                   for install() to use
         :type binary_pacakge: bool
+        :param njobs: number of jobs. It is mapped to make's -j option. If
+                      the value passed is of infinity type then make will not
+                      limit the jobs number.
+        :type njobs: int or infinity
         """
+        make_args = []
         LOG.info("Starting build the kernel")
-        build_output_format = ""
+
+        if njobs > 1:
+            make_args.append('-j')
+            if not math.isinf(njobs):
```

Comment:
I understand you need something unique here, but my taste is for simpler constructions, like -1 for an unlimited number of jobs. Anyway, I think that is a matter of personal choice.

Reply:
I thought in use negative integer to represent unlimited jobs but my brain refused to accept the idea of 'have -x jobs is equal to unlimited jobs'. And at same time using infinity is conceptually correct (I guess) but not usual. Maybe I could have used `None` to represent that?

Reply:
I don't think `None` makes more sense than an architecture defined constant, like `inf`. So, here, I believe the case is more aesthetic than a logic problem.

Reply:
Among my problem with `inf` is requiring the user to import from the math module.  Also, `make` describes the empty `-j` option as (https://www.gnu.org/software/make/manual/make.html#Parallel):

"If there is nothing looking like an integer after the ‘-j’ option, there is no limit on the number of job slots."

Which can, IMO, be read as the lack of a `-j` option, thus, `None` doesn't sound as bad.  So to summarize it:

a) No `-j` means default job slot size, which is 1
b) `-j` with an integer defines the specific job slot size
c) `-j` without an integer means no limits to the job slot size

If we want to follow `make`, I'd map this as:

a) `njobs=1` is the default parameter value
b) `njobs` with any other integer defines the specific job slot size
c) `njobs=None` means no limits to the job slot size

Now note that this behavior is different than `avocado.utils.build`.  We should decide if we want to follow make, or have a more aggressive parallelization by default.

Reply:
> Among my problem with `inf` is requiring the user to import from the math module. Also, `make` describes the empty `-j` option as (https://www.gnu.org/software/make/manual/make.html#Parallel):
>

You don't need to import `math` if you use `float('inf')`. Regardless, I got your point. 

> "If there is nothing looking like an integer after the ‘-j’ option, there is no limit on the number of job slots."
> 
> Which can, IMO, be read as the lack of a `-j` option, thus, `None` doesn't sound as bad. So to summarize it:
> 
> a) No `-j` means default job slot size, which is 1
> b) `-j` with an integer defines the specific job slot size
> c) `-j` without an integer means no limits to the job slot size
> 
> If we want to follow `make`, I'd map this as:
> 
> a) `njobs=1` is the default parameter value
> b) `njobs` with any other integer defines the specific job slot size
> c) `njobs=None` means no limits to the job slot size
> 

Looking from this angled `njobs=None` really doesn't sound as bad.

> Now note that this behavior is different than `avocado.utils.build`. We should decide if we want to follow make, or have a more aggressive parallelization by default.

In the case of `KernelBuild.build()` we can default to aggressive parallelization.

As for `avocado.utils.build` I've some doubts. There should be some projects  that not build with parallel jobs at all, and that module defaults to `multiprocessing.cpu_count() + 1`. Unless we assume that on those rare(?)  projects user *must* set `-j 1`. Anyway, `avocado.utils.build` default behaviour should be documented.

Reply:
I'm about to send a v2 where:
 * If njobs is None then '-j' is passed without value
 * default value of njobs is multprocessing.cpu_count()